pkg/router/template/router.go

```diff
@@ -86,13 +85,23 @@ type templateRouter struct {
 	statsPort int
 	// if the router should allow wildcard routes.
 	allowWildcardRoutes bool
-	// rateLimitedCommitFunction is a rate limited commit (persist state + refresh the backend)
-	// function that coalesces and controls how often the router is reloaded.
-	rateLimitedCommitFunction *ratelimiter.RateLimitedFunction
-	// rateLimitedCommitStopChannel is the stop/terminate channel.
-	rateLimitedCommitStopChannel chan struct{}
-	// lock is a mutex used to prevent concurrent router reloads.
+	// lock is a mutex used to prevent concurrent router state updates
 	lock sync.Mutex
+	// Track the start and end of router reloads
```

Comment:
Oh wow.  This is a LOT of variables.  This should be abstracted into some sort of separate object that is passed in because the rest of the template router has nothing to do with it.

Reply:
Please hide this complexity so that it doesn't get entangled with other places.